### PR TITLE
feat: add support for pagination

### LINKED ISSUE #5: Add support for pagination
### Description

GitHub supports pagination for several of it's apis: https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28#using-link-headers

I would like to add a `paginated_send` method that appears on the `Request` struct for all endpoints that return data which implements `IntoIterator`. 

It extracts the page data from the link header and returns the `PaginatedData` struct.

```rust
pub struct LinkedPages {
	pub previous: Option<i64>,
	pub next: Option<i64>,
	pub first: Option<i64>,
	pub last: Option<i64>,
}

pub struct PaginatedData<Data> {
	data: Data,
	pages: LinkedPages,
}
```

I'll push a PR soon.

### PR DESCRIPTION
**Pull request by ifiokjr**
Two new methods on the `Request` struct: `paginated_send` and `paginated_send_with_response`.

This also fixes the broken trait on `InstallationToken` which I missed in a previous PR. 

Fixes #5

### CODE REVIEW

**Comment by ifiokjr** [comment]
Once this passes tests it should be ready to merge. 

I've removed the reliance on `use crate::types::*;` which is why so many files have been changed.

File: octocrate/Cargo.toml [hunk 1]
```diff
@@ -66,42 +67,42 @@ dotenv = { workspace = true }
 [[example]]
 name = "commits"
 path = "examples/commits.rs"
-required-features = ["repos"]
+required-features = ["integration", "repos"]
```
Comment (panghu-huang) [inline octocrate/Cargo.toml:70]: It seems that `integration` is not listed in the features list, and the example does not require this feature

**Review by panghu-huang — CHANGES_REQUESTED** [review]
Can you add an example for pagination? That would make it more convenient to use.

I will try to fix the testing for `integration` (GitHub App / Installation Token) later.

**Comment by panghu-huang** [comment]
Thank you. I'll go ahead and merge this.

**Pull request merged**